### DISCUSSION TITLE
Update potBareSoilEvap in soil.py?

### DISCUSSION THREAD

**Discussion by Han-Su22** [iiasa/CWatM#34] [Q&A]
Dear all,

I am trying to add additional landcover to CWatM by adding some codes. During the processes, I found one place is confusing in soil.py, the line 257:

_self.var.potBareSoilEvap = np.maximum(0.,self.var.potBareSoilEvap - self.var.openWaterEvap[No])_

This will execute when the coverType is 'irrPaddy'. This line updates the potBareSoilEvap, which means for 'irrPaddy', some potBareSoilEvap is zero. This make sense for now. However, for 'irrNonPaddy', some potBareSoilEvap is also zero. Because in the landcoverType.py, the soil process is calculated for irrPaddy before irrNonPaddy. This means when calculating the soil process for  irrNonPaddy, the model use this updated potBareSoilEvap. From the results, the actBareSoilEvap is zero for irrNonPaddy where the irrPaddy has a larger open water evaporation. It seems does not make sense. Is this a bug?

Best,
Han Su

**Comment by Han-Su22** [comment]
Hi all,

I am almost sure this is a small bug. To confirm it:

1. Run the CWatM model with global 30min input (or any model with nonzero irrPaddy and irrNonPaddy input), add the _actBareSoilEvap[3], openWaterEvap[2]_ to the output _OUT_Map_Daily_
2. Open the results map. Choose any date, find a region where _openWaterEvap[2]_ is large (usually in the tropics), then the _actBareSoilEvap[3]_ is zero

This means when there is irrPaddy, the soil evaporation of irrNonPaddy was not taken into account.

To fix it, in the soil.py, I tried the following things:
1. Annotate (or delete) the line 257:

_self.var.potBareSoilEvap = np.maximum(0.,self.var.potBareSoilEvap - self.var.openWaterEvap[No])_

2. Replace the line 354:

_self.var.actBareSoilEvap[No] = np.minimum(self.var.potBareSoilEvap,np.maximum(0.,self.var.w1[No] - self.var.wres1[No]))_

with 

    if coverType == 'irrPaddy':

        self.var.actBareSoilEvap[No] = np.minimum(np.maximum(0.,self.var.potBareSoilEvap - self.var.openWaterEvap[No]),np.maximum(0.,self.var.w1[No] - self.var.wres1[No]))

    else:

        self.var.actBareSoilEvap[No] = np.minimum(self.var.potBareSoilEvap,np.maximum(0.,self.var.w1[No] - self.var.wres1[No]))


Please correct me if I am wrong.

**Reply by SongshGeo** [reply]
Mark, thanks for your work. I'll re-check it when I use this module.
